book/src/scalable_lists.md

```diff
@@ -0,0 +1,179 @@
+
+# // Please ignore this line
+# // It is only there to make mdbook happy
+# fn main() {}
+```
+
+We start be filling our models with the integers from 0 to 1000.
+
+<span class="filename">Filename: listings/scalable_lists/1/main.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/scalable_lists/1/main.rs:model}}
+```
+
+The `ListItemFactory` takes care of the widgets as well as their relationship to the model.
+Here we use the [`SignalListItemFactory`](../docs/gtk4/struct.SignalListItemFactory.html) which emits a signal for every relevant step in the life of a `ListItem`.
+The "setup" signal will be emitted when new widgets have to be created.
+We connect to it to create a `Label` for every requested widget.
+
+<span class="filename">Filename: listings/scalable_lists/1/main.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/scalable_lists/1/main.rs:factory_setup}}
+```
+
+In the "bind" step we bind the data in our model to the individual list items.
+
+<span class="filename">Filename: listings/scalable_lists/1/main.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/scalable_lists/1/main.rs:factory_bind}}
+```
+
+We only want single items to be selectable so we choose [`SingleSelection`](../docs/gtk4/struct.SingleSelection.html).
+The other options would have been [`MultiSelection`](../docs/gtk4/struct.MultiSelection.html) or [`NoSelection`](../docs/gtk4/struct.NoSelection.html).
+Then we pass the model and the factory to the [`ListView`](../git/docs/gtk4/struct.ListView.html).
+
+<span class="filename">Filename: listings/scalable_lists/1/main.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/scalable_lists/1/main.rs:selection_list}}
+```
+
+Every `ListView` has to be inside a [`ScrolledWindow`](../docs/gtk/struct.ScrolledWindow.html) so we are adding it to one.
+
+<span class="filename">Filename: listings/scalable_lists/1/main.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/scalable_lists/1/main.rs:scrolled_window}}
+```
+
+We can now easily scroll through our long list of integers.
+
+<div style="text-align:center"><img src="img/scalable_lists_demo_1.png"/></div>
+
+Let us see what else we can do.
+We might want to increase the number every time we activate its row.
+For that we first add the method `increase_number` to our `IntegerObject`.
+
+<span class="filename">Filename: listings/scalable_lists/2/integer_object/mod.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/scalable_lists/2/integer_object/mod.rs:integer_object}}
+```
+
+In order to interact with our `ListView`, we connect to its "activate" signal.
+
+<span class="filename">Filename: listings/scalable_lists/2/main.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/scalable_lists/2/main.rs:list_view_activate}}
+```
+
+Now every time we activate an element, for example by double-clicking on it, the corresponding "number" property of the `IntegerObject` in the model will be increased by 1.
```

Comment:
```suggestion
Now every time we activate an element, for example by double-clicking on it, the corresponding `number` property of the `IntegerObject` in the model will be increased by 1.
```

Reply:
I try to consistently use quotes instead of backticks when I talk about properties.